code/modules/games/cards.dm

```diff
@@ -0,0 +1,443 @@
+
+		var/colour
+		if(suit == "spades" || suit == "clubs")
+			colour = "black_"
+		else
+			colour = "red_"
+
+		for(var/number in list("ace","two","three","four","five","six","seven","eight","nine","ten"))
+			P = new()
+			P.name = "[number] of [suit]"
+			P.card_icon = "[colour]num"
+			P.back_icon = "card_back"
+			cards += P
+
+		for(var/number in list("jack","queen","king"))
+			P = new()
+			P.name = "[number] of [suit]"
+			P.card_icon = "[colour]col"
+			P.back_icon = "card_back"
+			cards += P
+
+	for(var/i = 0, i<2, i++)
+		P = new()
+		P.name = "joker"
+		P.card_icon = "joker"
+		cards += P
+
+/obj/item/weapon/deck/attackby(obj/O as obj, mob/user as mob)
+	if(istype(O,/obj/item/weapon/hand))
+		var/obj/item/weapon/hand/H = O
+		if(H.parentdeck == src)
+			for(var/datum/playingcard/P in H.cards)
+				cards += P
+			qdel(H)
+			to_chat(user,"<span class='notice'>You place your cards on the bottom of [src]</span>.")
+			return
+		else
+			to_chat(user,"<span class='warning'>You can't mix cards from other deck!</span>")
+			return
+
+	..()
+
+/obj/item/weapon/deck/attack_hand(mob/user as mob)
+	draw_card(user)
+
+/obj/item/weapon/deck/verb_pickup() // Speshul Snowflek Override for card so pickup work as intended.
+	if(!(usr)) //BS12 EDIT
+		return
+	if(!usr.canmove || usr.stat || usr.restrained() || !Adjacent(usr))
+		return
+	if((!istype(usr, /mob/living/carbon)) || (istype(usr, /mob/living/carbon/brain)))//Is humanoid, and is not a brain
+		to_chat(usr, "<span class='warning'>You can't pick things up!</span>")
+		return
+	if(usr.stat || usr.restrained() )//Is not asleep/dead and is not restrained
+		to_chat(usr, "<span class='warning'>You can't pick things up!</span>")
+		return
+	if(anchored) //Object isn't anchored
+		to_chat(usr, "<span class='warning'>You can't pick that up!</span>")
+		return
+	if(!usr.hand && usr.r_hand) //Right hand is not full
+		to_chat(usr, "<span class='warning'>Your right hand is full.</span>")
+		return
+	if(usr.hand && usr.l_hand) //Left hand is not full
+		to_chat(usr, "<span class='warning'>Your left hand is full.</span>")
+		return
+	if(!istype(src.loc, /turf)) //Object is on a turf
+		to_chat(usr, "<span class='warning'>You can't pick that up!</span>")
+		return
+	//All checks are done, time to pick it up!
+	usr.put_in_hands()
+	return
+
+/obj/item/weapon/deck/verb/draw_card(mob/user as mob)
+
+	set category = "Object"
+	set name = "Draw"
+	set desc = "Draw a card from a deck."
+	set src in view(1)
+
+	var/mob/living/carbon/human/M = user
+
+	if(user.stat || !Adjacent(user))
+		return
+
+	if(!cards.len)
+		to_chat(user,"<span class='notice'>There are no cards in the deck.</span>")
+		return
+
+	var/obj/item/weapon/hand/H = M.is_in_hands(/obj/item/weapon/hand)
+	if(H && !(H.parentdeck == src))
+		to_chat(user,"<span class='warning'>You can't mix cards from other deck!</span>")
+		return
+
+	if(!H)
+		H = new(get_turf(src))
+		user.put_in_hands(H)
+
+	if(!H || !user) return
```

Comment:
Two lines here. Also I'm not sure what you're trying to do here -- `!(H)` doesn't do anything because you already did this above:

```
if(!H)
	H = new(get_turf(src))
	user.put_in_hands(H)
```
and `!(user)` does nothing because the proc already needs a user